package.json

```diff
@@ -24,9 +24,10 @@
   ],
   "source": true,
   "scripts": {
+    "ts": "tsc --outDir ./build --declaration --declarationDir ./dist/types",
     "test": "mocha",
-    "build": "rollup -c",
-    "lint": "eslint --ignore-path .gitignore ."
+    "build": "rollup -c && yarn ts",
```

Comment:
An alternative here would be to run `tsc` first and then `rollup` the compiled files ... `tsc` is capable of generating es5/es2015 module files, and then `rollup` can use the `es5` output to build the umd ... I've used this approach a fair bit and it seems to be pretty solid :+1: